src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -905,23 +905,24 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
 {
     params.clear();
     uri.clear();
+    const auto parts = Utils::SplitString(string, L';');
+    if (parts.size() < 2)

```

Comment:
Would this not cause problems for URLs containing a semicolon? I'm assuming you'd get more than 2 parts, and you'd then just be dropping part of the URL. Or have I misunderstood the way this works?

Reply:
Interesting. Obviously no test case is covering the URLs containing a `;`. And yeah I think it's flawed. I'll add more cases.